src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -152,6 +152,15 @@ BlockIO InterpreterDropQuery::executeToTableImpl(ContextPtr context_, ASTDropQue
                 "Table {} is not a Dictionary",
                 table_id.getNameForLogs());
 
+        if (settings.ignore_drop_queries_probability != 0 && ast_drop_query.kind == ASTDropQuery::Kind::Drop && std::uniform_real_distribution<>(0.0, 1.0)(thread_local_rng) <= settings.ignore_drop_queries_probability)
+        {
+            ast_drop_query.sync = false;
+            if (table->getName() != "Memory" && table->getName() != "Join")
```

Comment:
```suggestion
            if (table->storesDataOnDisk())
```

Reply:
Also let's add a log message

Reply:
`storesDataOnDisk ` returns true for `Join` table engine, AFAIU because it also writes data into backup file to restore it after restart. Do we want to truncate `Join` tables?

Reply:
@tavplubix WDYT?

Reply:
Hm, I think it makes sense to keep Join tables since they actually store data on disk and reload it on restart. If we want to reduce the memory consumption, we can implement unloading and reloading in runtime (it may be useful not only for testing)